### ISSUE TITLE
NgUpgrade Performance Issues

### ISSUE THREAD

**Issue by tcrite** [kamilkp/angular-vs-repeat#204]
Hi I'm working on a project to upgrade an AngularJs app to Angular 5 using ngupgrade. I had vs-repeat installed prior to the upgrade and it worked fined. However, since installing Angular 5 and bootstrapping Angular and AngularJs accordingly, I'm getting extremely poor performance from vs-repeat. Almost to the point where it's unusable. Was wondering if it could have something to do with AngularJs and Angular competing over change detection? Thanks and any help would be much appreciated.

**Comment by m3Lith** [comment]
I'm having the same issue currently. Did you manage to do something by any chance?

**Comment by tcrite** [comment]
@m3Lith if I comment this line out in my polyfills.ts file:

import 'zone.js/dist/zone';  // Included with Angular CLI.

it works ok. haven't had any ill effects from it yet, but I get the feeling it's not ideal

**Comment by m3Lith** [comment]
@tcrite but above that it says "Zone JS is required by default for Angular itself." and it throws an error inside console: "Error: In this configuration Angular requires Zone.js".

Does it really work for you with that one change?

**Comment by tcrite** [comment]
@m3Lith you are right. I just realized I added an import for zone.js in my main.ts file. Seems. to fix the configuration error thrown when you comment it out in the polyfills file

import 'zone.js';

**Comment by m3Lith** [comment]
Have you tried uncommenting/adding this line in polyfills.ts?
`(window as any).__Zone_disable_requestAnimationFrame = true;`
It didn't help me, but I wonder if you would see a difference.

**Comment by jziggas** [comment]
I don't have a polyfills.ts in my project so I'm not sure where that would go.

**Comment by m3Lith** [comment]
You're including zone.js somewhere though? Try adding that line before the import of it.

**Comment by davidfregoli** [comment]
Just encountered this issue, I can confirm @m3Lith's suggestion works

**Comment by amalitsky** [comment]
As of now _angular-vs-repeat_ virtually doesn't work in angularJS/angular hybrid mode due to constant `$digest` re-triggering.

@kamilkp, would this be considered as a reasonable fix in the form of PR? Thanks

```
function angularHybridModeIsOn() {
  return '$$UpgradeModule' in $element.injector().modules;
}

$scope.$watch(function () {
  if (typeof window.requestAnimationFrame === 'function' && !angularHybridModeIsOn()) { 
    window.requestAnimationFrame(reinitOnClientHeightChange);
  } else {
    reinitOnClientHeightChange();
  }
});
```
